gen/enum.go

```diff
@@ -68,5 +56,43 @@ func (g *Generator) enum(spec *compile.EnumSpec) error {
 		}{Spec: spec, Reader: typeReader(spec)},
 	)
 
+	if err == nil {
+		err = enumFromWire(g, spec)
+	}
+
 	return wrapGenerateError(spec.Name, err)
 }
+
+func enumFromWire(g *Generator, spec *compile.EnumSpec) error {
+	// Mapping from enum value to item name. In case of multiple items with the
+	// same value, any one of those could be set here.
+	valuesToNames := make(map[int32]string, len(spec.Items))
+	for _, item := range spec.Items {
+		valuesToNames[item.Value] = item.Name
+	}
+
+	return g.DeclareFromTemplate(
+		`
+			<$fmt := import "fmt">
+			<$wire := import "github.com/thriftrw/thriftrw-go/wire">
+
+			<$enumName := defName .Spec>
+			<$v := newVar "v">
+			<$w := newVar "w">
+
+			func (<$v> *<$enumName>) FromWire(<$w> <$wire>.Value) error {
+				switch <$w>.GetI32() {
```

Comment:
Can't we just do
```
(<$enumName>)(value)
```

That will handle unknown enums + doesn't care if there's 2 names for the same enum (both will work when the user tries to compare it etc)